pkg/proxy/ipvs/proxier.go

```diff
@@ -1693,23 +1693,26 @@ func (proxier *Proxier) cleanLegacyService(activeServices map[string]bool, curre
 		if _, ok := activeServices[cs]; !ok {
 			rsList, _ := proxier.ipvs.GetRealServers(svc)
 
-			// If we still have real servers graceful termination is not done
-			if len(rsList) > 0 {
```

Comment:
Am I correct that this if statement meant either we never attempted to execute the for loop, or when we did, it was iterating over a list of length 0?

Reply:
yeah, we never attempted to execute the for loop